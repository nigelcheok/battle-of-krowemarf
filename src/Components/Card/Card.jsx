import React from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

export function Card(props) {
  return (
    <div className="card border-square">
      {/*<img src="..." className="card-img-top" alt="...">*/}
        <div className="card-body" style={CardBodyStyle()}>
          <FontAwesomeIcon icon="coffee" />
          <h5 className="card-title" style={CardTitleStyle()}>{ props.title }</h5>
          <p className="card-text" style={CardDescriptionStyle()}>{ props.description }</p>
          {/*<a href="#" className="btn btn-primary">Go somewhere</a>*/}
        </div>
      <div className="card-footer" style={CardFooterStyle()}>
        test
      </div>
    </div>
  );
}

function CardBodyStyle() {
  return {
    backgroundColor: 'rgba(37,39,41,.01)',
  }
}

function CardTitleStyle() {
  return {
    fontSize: '1.25rem',
  }
}

function CardDescriptionStyle() {
  return {
    fontSize: '.9rem',
    color: 'rgb(142, 142, 143)',
  }
}

function CardFooterStyle() {
  return {
    borderTop: '0',
  }
}
