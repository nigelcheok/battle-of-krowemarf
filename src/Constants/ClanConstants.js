const allClans = [
  {
    clanName: 'React',
    clanIcon: 'fa-react',
    ownerName: 'facebook',
    repoName: 'react',
  },
  {
    clanName: 'Vue',
    clanIcon: 'fa-vuejs',
    ownerName: 'vuejs',
    repoName: 'vue',

  },
  {
    clanName: 'Angular',
    clanIcon: 'fa-angular',
    ownerName: 'angular',
    repoName: 'angular.js',
  }
];

export default {
  allClans
}