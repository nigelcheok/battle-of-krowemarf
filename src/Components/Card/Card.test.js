import React from 'react';
import { shallow } from 'enzyme';
import { Card } from './Card';
import ClanConstants from '../../Constants/ClanConstants';

describe('Card', () => {
  it('should render correctly with no props', () => {
    const component = shallow(<Card/>);
    expect(component).toMatchSnapshot();
  });

  it('should render correctly with props', () => {
    const clanDetailsData = {
      clan: ClanConstants.allClans[0],
      id: undefined,
      name: 'react',
      starsCount: 1,
      forksCount: 2,
      issuesCount: 3,
      pullRequestsCount: 4,
    };

    const component = shallow(<Card title="test" description="test description" clanDetails={clanDetailsData}/>);
    expect(component).toMatchSnapshot();
  });
});