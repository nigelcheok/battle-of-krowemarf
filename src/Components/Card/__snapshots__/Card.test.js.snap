// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should render correctly with no props 1`] = `<div />`;

exports[`Card should render correctly with props 1`] = `
<div>
  <Link
    style={
      Object {
        "color": "rgb(37,39,41)",
        "textDecoration": "none",
      }
    }
    to="/react"
  >
    <div
      className="card border-square mt-3 shadow-on-hover"
      style={Object {}}
    >
      <div
        className="card-body d-flex"
        style={
          Object {
            "backgroundColor": "rgba(37,39,41,.01)",
          }
        }
      >
        <div
          className="align-self-center mr-3 bg-white d-flex justify-content-center position-relative"
          style={
            Object {
              "borderRadius": "30px",
              "height": "60px",
              "width": "60px",
            }
          }
        >
          <div
            className="vertical-align"
          >
            <div
              className="fab fa-react fa-3x"
              style={
                Object {
                  "color": "#00d8ff",
                }
              }
            />
          </div>
        </div>
        <div>
          <h5
            className="card-title mt-1"
            style={
              Object {
                "fontSize": "1.25rem",
                "marginBottom": "0",
              }
            }
          >
            test
          </h5>
          <p
            className="card-text mt-1"
            style={
              Object {
                "color": "rgb(142, 142, 143)",
                "fontSize": ".9rem",
              }
            }
          >
            test description
          </p>
        </div>
      </div>
      <CardFooter
        clanDetails={
          Object {
            "clan": Object {
              "clanColor": "#00d8ff",
              "clanIcon": "fa-react",
              "clanName": "React",
              "ownerName": "facebook",
              "repoName": "react",
            },
            "forksCount": 2,
            "id": undefined,
            "issuesCount": 3,
            "name": "react",
            "pullRequestsCount": 4,
            "starsCount": 1,
          }
        }
      />
    </div>
  </Link>
</div>
`;
